src/search/headerBar.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from 'react';
+import {
+  View,
+  SafeAreaView,
+  TextInput,
+  Text,
+  TouchableHighlight,
+} from 'react-native';
+import { useDispatch } from 'react-redux';
+import { useNavigation } from 'react-navigation-hooks';
+import styles from './styles';
+import { searchData } from '../redux/search/actionCreater';
+
+export default function SearchBar() {
+  const [value, onChangeText] = useState('');
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+
+  useEffect(() => {
+    let timer;
+    if (value.length >= 2) {
+      timer = setTimeout(() => {
+        dispatch(searchData(value));
+      }, 300);
+    }
+    return () => {
+      if (timer) clearTimeout(timer);
+    };
```

Comment:
Is this the suggested place for this? Or should it go the onChnage of the field? I dont know, im just asking, pls try to investiagte. Im okay with this, just pls give me a reason why is it here.

Reply:
We want to implement function for auto search, trying to listen the change on input. I have tried to put it in onChange function, but it seems not work properly.